src/libguarded/guarded.hpp

```diff
@@ -19,6 +19,48 @@
 namespace libguarded
 {
 
+template <typename T, typename D>
+struct Handle
+{
+    Handle(T* t, D& d) :
+        ref(*t),
```

Comment:
by dereferencing t, you hit undefined behaviour if t is nullptr. the ref member variable should be stored as a pointer, not a reference